src/search/view/SearchListView.js

```diff
@@ -127,14 +124,38 @@ export class SearchListView {
 			fetch: (startId, count) => {
 				if (locator.search.indexState().initializing) {
 					// show spinner until the actual search index is initialized
-					return defer().promise
+					return new Promise(noOp)
 				}
-				if (!this._searchResult || this._searchResult.results.length === 0 && !hasMoreResults(this._searchResult)) {
+
+				const lastResult = this._searchResult
+				if (!lastResult || lastResult.results.length === 0 && !hasMoreResults(lastResult)) {
 					return Promise.resolve([])
 				}
-				return this._loadSearchResults(this._searchResult, startId !== GENERATED_MAX_ID, startId, count)
-				           .then(results => results.map(instance => new SearchResultListEntry(instance)))
-				           .finally(m.redraw)
+
+
+				// If search is triggered and completes again before we finished loading the results from the previous search
+				// then we will end up loading results again and they will be inserted into the list, resulting in duplicate entries
+				// this can happen if the user hits the enter key fast while in the search bar
+				// so we will ignore whatever is being downloaded from the last search, and the new search will update the list
+				if (this._lastSearchResults) {
+					this._lastSearchResults.resolve([])
+					this._lastSearchResults = null
+				}
+
+				// save to a local var because flow complains at about it being nullable when returning
+				const deferredResult = defer()
+				this._lastSearchResults = deferredResult
+
+				this._loadSearchResults(lastResult, startId !== GENERATED_MAX_ID, startId, count)
+		                    .then(results => results.map(instance => new SearchResultListEntry(instance)))
+				    .then(results => {
+					    if (this._lastSearchResults) {
```

Comment:
What if this is not our deferred but already a new one? I think check should be `this._lastSearchResults === deferredResult` and only then we should resolve it?